src/celeritas/phys/PhysicsStepUtils.hh

```diff
@@ -180,35 +156,25 @@ calc_physics_step_limit(const MaterialTrackView& material,
  * interpolation.
  *
  * Zero energy loss can occur in the following cases:
- * - The particle doesn't have slowing-down energy loss (e.g. photons)
  * - The energy loss value at the given energy is zero (e.g. high energy
  * particles)
  * - The urban model is selected and samples zero collisions (possible in thin
  * materials and/or small steps)
  */
-template<class Engine>
-CELER_FUNCTION ParticleTrackView::Energy
-               calc_energy_loss(const CutoffView&        cutoffs,
-                                const MaterialTrackView& material,
-                                const ParticleTrackView& particle,
-                                const PhysicsTrackView&  physics,
-                                real_type                step,
-                                Engine&                  rng)
+inline CELER_FUNCTION ParticleTrackView::Energy
+                      calc_mean_energy_loss(const ParticleTrackView& particle,
+                                            const PhysicsTrackView&  physics,
+                                            real_type                step)
 {
     CELER_EXPECT(step > 0);
+    CELER_EXPECT(physics.eloss_ppid());
     using Energy = ParticleTrackView::Energy;
     using VGT    = ValueGridType;
     static_assert(Energy::unit_type::value()
                       == EnergyLossCalculator::Energy::unit_type::value(),
                   "Incompatible energy types");
 
     auto ppid = physics.eloss_ppid();
-    if (!ppid)
-    {
-        // No energy loss processes for this particle
-        return Energy{0};
-    }
-
     const real_type pre_step_energy = value_as<Energy>(particle.energy());
```

Comment:
Since this is based on the currently, we do not have the approximation below the lowest kinetic energy (i.e, just returning eloss = particle.energy()).  Will it be fixed up later or are you going to actually drop it or move it somewhere else?

Reply:
Yep, when I merged master yesterday (after merging your work) this should be fixed.